### PR TITLE
FEATURE: prioritize the user who is getting the reply in the autocomplete

### PR DESCRIPTION
**Pull request by Grubba27**
When replying to a post, the user who is getting the reply should be prioritized in the autocomplete

- Added in composer a getter for getting `.replyingToUser`
- Added in d-editor the reference to the user that is getting the reply(`this.composer.replyingToUser`)
- Passed along the reference to the user that is getting the reply to the user-search service as `replyingToUser`
- Controller `users_controller.rb` was modified to accept the `user_id` parameter and pass it to the `UserSearch` model
- The `UserSearch` model was modified to accept the `user_id` parameter and use it to prioritize the user that is getting the reply in the autocomplete on the first time you call the autocomplete service and while the username is included in the searched term
- Had to update the serializer to pass the id of the `replyingToUser` from the post

### CODE REVIEW

**Review by markvanlan — COMMENTED** [review]
My only concern is the naming of `user_id` for the user we are prioritizing. It reads kind of like we are passing in the current_user, not the user we are replying too.. What do you think about that?

**Comment by Grubba27** [comment]
The funny thing is that the `current_user` in this context is `searching_user`, I guess I could call it `prioritized_user_id`?

File: app/serializers/post_serializer.rb [hunk 1]
```diff
@@ -298,6 +298,7 @@ def trust_level
 
   def reply_to_user
     {
+      id: object.reply_to_user.id,
       username: object.reply_to_user.username,
```
Comment (pmusaraj) [inline app/serializers/post_serializer.rb:302]: Minor, but you could pass `username` instead of `id` in the `d-editor` component and with that you wouldn't need to add a new attribute to this serializer.

**Review by pmusaraj — APPROVED** [review]
Looks good to me!

File: app/serializers/post_serializer.rb (hunk 1 above)
Comment (Grubba27) [reply app/serializers/post_serializer.rb:302]: I feel that using a username could feel a bit weird–most places pass just the `user_id` for search.

And I have a strong belive that this specific id is not a secret(I was too security concerned)

Comment (pmusaraj) [reply app/serializers/post_serializer.rb:302]: Yeah, no security concerns with the ID. We include it in the basic user serializer. 

Ok to proceed with the `id`.

**Review by renato — APPROVED** [review]

**Review by tyb-talks — APPROVED** [review]
lgtm too, my only initial concern was around the edge case of when replyingToUser is null which is already handled well throughout the PR + existing test cases would exercise that behaviour.
